cmd/deploy.go

```diff
@@ -354,6 +355,67 @@ func autofillApimodel(dc *deployCmd) error {
 			}
 		}
 	}
+
+	if k8sConfig != nil && k8sConfig.Addons != nil && k8sConfig.IsContainerMonitoringAddonEnabled() {
+		log.Infoln("container monitoring addon enabled")
```

Comment:
nit: can you please refactor the container monitoring addon setup to its own function since the `autofillApimodel` is already overloaded? Then you'd have something like:

```
if k8sConfig != nil && k8sConfig.Addons != nil && k8sConfig.IsContainerMonitoringAddonEnabled() {
      err = setupMonitoringAddon(k8sConfig)
      if err != nil {
           return errors.Wrap(err, "Failed to setup container monitoring")
      }
} 
```

This will help make the deploy cmd code hopefully little more readable and also allow you to isolate the monitoring addon stuff for unit testing.

Reply:
(feel free to come up with a better name than setupMonitoringAddon, but you get the idea)

Reply:
> (feel free to come up with a better name than setupMonitoringAddon, but you get the idea)

Thanks for the feedback. Refactored the code as suggested.